src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -18,7 +19,11 @@ namespace ErrorCodes
 }
 
 IFileCachePriority::Iterator LRUFileCachePriority::add(
-    const Key & key, size_t offset, size_t size, std::weak_ptr<KeyMetadata> key_metadata)
+    const Key & key,
```

Comment:
Similar comment as to `Entry`, - key_ param is redundant here, but in general, I wonder why we don't store FileSegmentPtr in entry. It'd be more natural.